Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc

```diff
@@ -13,179 +13,89 @@
 {
-	_ddd			= ref._ddd;
-	_params			= ref._params;
-	_trans			= ref._trans;
-	_rot			= ref._rot;
-	_shape			= ref._shape;
-	_name			= ref._name;
-	_parents		= ref._parents;
-	_volume			= ref._volume;
-	_density		= ref._density;
-	_weight			= ref._weight;
-	_copy			= ref._copy;
-	_material		= ref._material;
-	_geographicalID	= ref._geographicalID;
-	_type			= ref._type;
-
-	return (*this);
+  m_params = ref.m_params;
+  m_trans = ref.m_trans;
+  m_rot = ref.m_rot;
+  m_name = ref.m_name;
+  m_copy = ref.m_copy;
+  m_geographicalID = ref.m_geographicalID;
+  m_z = ref.m_z;
+  return (*this);
 }
 
 //----------------------------------------------------------------------------------------------------
 
 DetGeomDesc::~DetGeomDesc()
 {
-	deepDeleteComponents();
+  deepDeleteComponents();
 }
 
 //----------------------------------------------------------------------------------------------------
 
-DetGeomDesc::Container DetGeomDesc::components()
+DetGeomDesc::Container DetGeomDesc::components() const
 {
-	return _container;
+  return m_container;
 }
 
 //----------------------------------------------------------------------------------------------------
 
-DetGeomDesc::ConstContainer DetGeomDesc::components() const
+void DetGeomDesc::addComponent( DetGeomDesc* det )
 {
-	ConstContainer _temp;
-	for (auto it : _container) {
-		_temp.emplace_back(it);
-	}
-	return _temp;
-}
-
-//----------------------------------------------------------------------------------------------------
-
-DetGeomDesc::ConstContainer DetGeomDesc::deepComponents() const
-{
-  ConstContainer _temp;
-  if (isLeaf())
-    _temp.emplace_back(const_cast<DetGeomDesc*>(this));
-  else {
-    for (auto it : _container){
-      ConstContainer _temp2 =  (*it).deepComponents();
-      copy(_temp2.begin(), _temp2.end(), back_inserter(_temp));
-    }
-  }
-  return _temp;
-}
-
-
-//----------------------------------------------------------------------------------------------------
-
-void DetGeomDesc::addComponents(Container cont)
-{
-	for(auto & ig : cont) {
-		_container.emplace_back(ig);
-	}
-}
-
-//----------------------------------------------------------------------------------------------------
-
-void DetGeomDesc::addComponent(DetGeomDesc* det)
-{
-	_container.emplace_back(det);
+  m_container.emplace_back( det );
 }
 
 //----------------------------------------------------------------------------------------------------
 
 void DetGeomDesc::deleteComponents()
 {
-	_container.erase(_container.begin(), _container.end());
+  m_container.erase( m_container.begin(), m_container.end());
 }
 
 //----------------------------------------------------------------------------------------------------
 
 void DetGeomDesc::deepDeleteComponents()
 {
-	for (auto & it : _container) {
-		( const_cast<DetGeomDesc*>(it) )->deepDeleteComponents();
-		delete it;
-	}
-	clearComponents();  
+  for( auto & it : m_container ) {
+    ( const_cast<DetGeomDesc*>(it) )->deepDeleteComponents();
```

Comment:
Why is the cast necessary? Aren't the elements of m_container DetGeomDesc pointers?

Reply:
done